### PR TITLE
move chainConfigInformation to config.genesis and update build_chain

### PR DESCRIPTION
**Pull request by wenlinlee**
…tools

### CODE REVIEW

**Review by ywy2090 — APPROVED** [review]

File: bcos-framework/bcos-framework/ledger/genesisConfig.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        std::stringstream ss;
+        ss << "[chian]" << std::endl
+           << "sm_crypto:" << m_smCrypto << std::endl
+           << "chainID: " << m_chainID << std::endl
+           << "grouID: " << m_groupID << std::endl
+           << "[consensys]" << std::endl
+           << "consensus_type: " << m_consensusType << std::endl
+           << "block_tx_count_limit:" << m_blockLimit << std::endl
+           << "leader_period:" << m_leaderSwitchPeriod << std::endl
```
Comment (JimmyShi22) [inline bcos-framework/bcos-framework/ledger/genesisConfig.h:66]: missing node.1 node.2 ?

File: bcos-tool/bcos-tool/NodeConfig.cpp [hunk 2]
```diff
@@ -56,7 +57,12 @@ NodeConfig::NodeConfig(KeyFactory::Ptr _keyFactory)
 
 void NodeConfig::loadConfig(boost::property_tree::ptree const& _pt, bool _enforceMemberID)
 {
-    loadChainConfig(_pt);
+    // if version < 3.1.0, config.ini include chianConfig
+    if (m_compatibilityVersion < (uint32_t)bcos::protocol::Version::V3_1_VERSION &&
```
Comment (JimmyShi22) [inline bcos-tool/bcos-tool/NodeConfig.cpp:61]: Please make sure that loadConfig() is run after loadGenesisConfig()
For example, in ExecutorServiceApp.cpp:140, loadGenesisConfig is running after loadConfig, so we can't get m_compatibilityVersion in loadConfig

Comment (JimmyShi22) [reply bcos-tool/bcos-tool/NodeConfig.cpp:61]: storageTool.cpp
AirNodeInitializer.cpp
SchedulerServiceApp.cpp
Initializer.cpp
lightnode/main.cpp

Comment (JimmyShi22) [reply bcos-tool/bcos-tool/NodeConfig.cpp:61]: Please add ut to make sure that logic

File: bcos-framework/bcos-framework/ledger/genesisConfig.h [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+#include "bcos-ledger/src/libledger/Ledger.h"
+#include <sstream>
+#include <string>
+
+
+namespace bcos
+{
+namespace ledger
+{
+class genesisConfig
```
Comment (JimmyShi22) [inline bcos-framework/bcos-framework/ledger/genesisConfig.h:34]:
```suggestion
class GenesisConfig
```

File: bcos-tool/bcos-tool/NodeConfig.cpp [hunk 4]
```diff
@@ -757,6 +771,27 @@ void NodeConfig::generateGenesisData()
...
+            m_isAuthCheck, m_authAdminAddress, m_isSerialExecute);
+        genesisdata = genesisData->genesisDataOutPut();
+        size_t j = 0;
+        for (const auto& node : m_ledgerConfig->consensusNodeList())
+        {
+            genesisdata = genesisdata + "node." + boost::lexical_cast<std::string>(j) + ":" +
+                          *toHexString(node->nodeID()->data()) + "," +
+                          std::to_string(node->weight()) + "\n";
+            ++j;
+        }
```
Comment (JimmyShi22) [inline bcos-tool/bcos-tool/NodeConfig.cpp:789]: node.1 node2. ... can be added in class GenesisConfig

File: libinitializer/Initializer.cpp [hunk 5]
```diff
@@ -95,8 +95,8 @@ void Initializer::initConfig(std::string const& _configFilePath, std::string con
     std::string const& _privateKeyPath, bool _airVersion)
 {
     m_nodeConfig = std::make_shared<NodeConfig>(std::make_shared<bcos::crypto::KeyFactoryImpl>());
-    m_nodeConfig->loadConfig(_configFilePath);
     m_nodeConfig->loadGenesisConfig(_genesisFile);
+    m_nodeConfig->loadConfig(_configFilePath);
```
Comment (JimmyShi22) [inline libinitializer/Initializer.cpp:99]: That's right, please check in other files

**Review by JimmyShi22 — APPROVED** [review]

**Pull request merged**